rules/xcodeproj.bzl

```diff
@@ -506,6 +506,10 @@ Double check your rule declaration for naming or add `xcodeproj-ignore-as-target
         target_settings["SWIFT_ACTIVE_COMPILATION_CONDITIONS"] = " ".join(
             ["\"%s\"" % d for d in defines_without_equal_sign],
         )
+        target_settings["BAZEL_LLDB_SWIFT_EXTRA_CLANG_FLAGS"] = " ".join(
```

Comment:
this is the key for getting swift to work again right?

Reply:
There are two issues: (1) LLDB settings need to be right or else LLDB will dump a bunch of errors and (2) the linker command for the application/test needs to include the `add_ast_path` option. This linker option needs to get propagated down to the application/test target by setting `linkopts` in the ObjcProvider returned from the framework target for the dependency.

Reply:
thx for this! I should have caught them on the description section. 
Looking back, I wonder how can we know these are the steps needed without Sam's input...
Also did you manually ensure that breakpoints work in test projects in this repo? If not that's fine I can do another round of tests

Reply:
Sure, I can explain how I went about finding these problems. I first added the line `log enable lldb types -f /tmp/types.log` to my lldbinit, so that I would get very verbose diagnostic lldb logging in a file called /tmp/types.log

1. I noticed the linker invocations being made by bazel to generate the final binary did not reference all the AST paths for transitive dependencies. I saw that LLDB was (1) sometimes complaining about missing ASTs and (2) in the types.log file , I noticed that the AST for transitive deps were not being loaded (They were being loaded in the working case.) So I just experimented with adding on the `add_ast_path` option to the linker and that did the trick.

As far as the actual implementation of this change, Sam noticed that framework.bzl was not propagating linker opts for transitive deps, which would explain why these linker options were not present.

2. After I made that fix, debugging was still not working. The LLDB output was complaining about an error with importing GPBProtocolBuffers.h. I noticed that one of our projects at Square has a clang flag with GPB in the name (this flag affects the way we build source files that depend on google protobufs.) So I used the lldb setting for extra-clang-flags to add all the necessary clang flags.

Reply:
I will double check the breakpoints in the test project, but note that debugging in the test project only works when sandboxing is disabled (which requires changing local bazelrc.)

In order to test this change, I updated build.sh to check for the presence of AST files in the symbol table for the binary. I think that should be sufficient.

Reply:
Thanks for the detailed explanation! Learnt that we can do logging for lldb just like we can do it for SourceKIT (so in future get the logs is the priority so we can not shooting in the dark)
As far as the testing go, I think it might be worth to call out in readme that sandbox needs to be disabled to get debugging working? Definitely a separate concern tho

Reply:
@amberdixon Can point to me where are the changes on build.sh for presence of AST files? 🙏

Reply:
> @amberdixon Can point to me where are the changes on build.sh for presence of AST files? 🙏

saw the updated checks, that's awesome!